javaagent-extension-api/src/main/java/io/opentelemetry/javaagent/extension/instrumentation/InstrumentationModule.java

```diff
@@ -117,11 +117,19 @@ public boolean isHelperClass(String className) {
     return false;
   }
 
-  /** Returns a list of resource names to inject into the user's classloader. */
+  /**
+   * Returns a list of resource names to inject into the user's class loader.
+   *
+   * @deprecated use {@link #registerHelperResources(HelperResources)} instead.
+   */
+  @Deprecated
   public List<String> helperResourceNames() {
     return Collections.emptyList();
   }
 
+  /** Register resource names to inject into the user's class loader. */
+  public void registerHelperResources(HelperResources helperResources) {}
```

Comment:
better name for `HelperResources`?

Reply:
`HelperResourcesBuilder`? not sure if it's better though 😄